speechbrain/core.py

```diff
@@ -1281,6 +1288,38 @@ def update_average(self, loss, avg_loss):
             avg_loss += float(loss) / self.step
         return avg_loss
 
+    @contextmanager
+    def no_sync(self, use=True):
+        """Copies pytorch's implementation for doing no_sync across all modules.
+
+        Explanation: nn.module.no_sync() is a context manager for when one does
+        not want to sync gradients, which happens when using both DDP and gradient accumulation.
+        Speechbrain brain's class can contain multiple modules and calling no_sync on these
+        individually would be very awkward, therefore this contextmanager exists.
```

Comment:
This seems like a useful utility! I'm curious though, is there a reason we can't use the PyTorch version of `no_sync` and have to roll our own?

Reply:
Imagine your brain class has modules: `[module1, module2, module3]`

Then you would have to call:
```
with module1.no_sync():
    with module2.no_sync():
        with module3.no_sync():
            fit_batch contents
```
which I don't think is nice..

Reply:
Aha, this makes sense